app/src/main/java/ayds/lisboa/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,228 @@
+        updateView(artistInfo)
+    }
+
+    private fun getArtistInfo(artistName : String): ArtistInfo{
+        var artistInfo = getArtistBioFromDatabase(artistName)
+        if (artistInfo.info != null)
+            markArtistInfoAsLocalStored(artistInfo)
+        else {
+            artistInfo = getArtistBioFromAPI(artistName)
+            if (artistInfo.info != null)
+                saveBio(artistName, artistInfo)
+        }
+        return artistInfo
+    }
+
+    private fun updateView(artistInfo: ArtistInfo){
+        updateViewUrl(artistInfo)
+        updateViewInfo(artistInfo)
+    }
+
+    private fun updateViewUrl(artistInfo: ArtistInfo){
+        val url = artistInfo.url
+        url?.let { setListenerURL(it) }
+    }
+
+    private fun updateViewInfo(artistInfo: ArtistInfo){
+        val info = getInfoFromArtistInfo(artistInfo)
+        textToHtml(info)
+        setTextInfoView(info)
+    }
+
+    private fun getInfoFromArtistInfo(artistInfo: ArtistInfo): String{
+        var info = artistInfo.info
+        val isLocallyStored = artistInfo.isLocallyStored
+        if (isLocallyStored)
+            info = "[*]$info"
+        else if (info == null)
+            info = NO_RESULTS
+        return info
+    }
+
+    private fun getArtistBioFromAPI(artistName: String): ArtistInfo {
+        val artist = getArtistAPI(artistName)
+        val bio = getBioFromArtist(artist)
+        val info = formatInfo(bio, artistName)
+        val url = getURLFromArtist(artist)
+        return ArtistInfo(url, info, false)
+    }
+
+    private fun markArtistInfoAsLocalStored(artistInfo: ArtistInfo) {
+        artistInfo.isLocallyStored = true
+    }
+
+    private fun getArtistBioFromDatabase(artistName: String): ArtistInfo {
+        val info = dataBase.getInfo(artistName)
+        return ArtistInfo(null, info, info != null)
+    }
+
+    private fun getArtistAPI(artistName: String?): JsonObject? {
+        val callResponse = getSongFromService(artistName)
+        return getSongFromExternalData(callResponse.body())
+    }
+
+    private fun getSongFromExternalData(serviceData: String?): JsonObject? {
+        val gson = Gson()
+        val jobjBody = gson.fromJson(serviceData, JsonObject::class.java)
+        val jobjArtist = jobjBody[JSON_ARTIST]
+        return jobjArtist.asJsonObject
+    }
+
+    private fun getSongFromService(artistName: String?): Response<String> {
+        return lastFMAPI.getArtistInfo(artistName).execute()
+    }
+
+    private fun getBioFromArtist(artist: JsonObject?): String? {
+        val bio = artist?.get(JSON_BIO)
+        val bioJson = bio?.asJsonObject
+        return when(val content = bioJson?.get(JSON_CONTENT)) {
+            null -> null
+            else -> content.asString
+        }
+    }
+
+    private fun getURLFromArtist(artist: JsonObject?): String? =
+        when(artist) {
+            null -> null
+            else -> artist[JSON_URL].asString
+        }
```

Comment:
Se puede simplificar, `null ->  null` es un smell:
```suggestion
    private fun getURLFromArtist(artist: JsonObject?): String? = artist?.let { it[JSON_URL].asString }
```